docs/tutorials/go-built-in.md

```diff
@@ -10,44 +10,45 @@ This guide is designed for beginners who want to get started with a Tendermint
 Core application from scratch. It does not assume that you have any prior
 experience with Tendermint Core.
 
-Tendermint Core is Byzantine Fault Tolerant (BFT) middleware that takes a state
-transition machine - written in any programming language - and securely
-replicates it on many machines.
-
-Although Tendermint Core is written in the Golang programming language, prior
-knowledge of it is not required for this guide. You can learn it as we go due
-to it's simplicity. However, you may want to go through [Learn X in Y minutes
-Where X=Go](https://learnxinyminutes.com/docs/go/) first to familiarize
+Tendermint Core is a service that provides a Byzantine Fault Tolerant consensus engine
+for state-machine replication. The replicated state-machine, or "application", can be written
+in any language that can send and receive protocol buffer messages in a client-server model.
+Applications written in Go can also use Tendermint as a library and run the service in the same 
+process as the application. 
+
+By following along this tutorial you will create a Tendermint Core application called kvstore, 
+a (very) simple distributed BFT key-value store.
+The application will be written in Go and  
+some understanding of the Go programming language is expected.
+If you have never written Go, you may want to go through [Learn X in Y minutes
+Where X=Go](https://learnxinyminutes.com/docs/go/) first, to familiarize
 yourself with the syntax.
 
-By following along with this guide, you'll create a Tendermint Core project
-called kvstore, a (very) simple distributed BFT key-value store.
+Note: Please use the latest released version of this guide and of Tendermint.
+We strongly advise against using unreleased commits for your development.
 
-## Built-in app vs external app
 
-Running your application inside the same process as Tendermint Core will give
-you the best possible performance.
+### Built-in app vs external app
 
-For other languages, your application have to communicate with Tendermint Core
-through a TCP, Unix domain socket or gRPC.
+On the one hand, to get maximum performance you can run your application in 
+the same process as the Tendermint Core, as long as your application is written in Go. 
+[Cosmos SDK](https://github.com/cosmos/cosmos-sdk) is written
+this way. 
+This is the approach followed in this tutorial.
 
-## 1.1 Installing Go
-
-Please refer to [the official guide for installing
-Go](https://golang.org/doc/install).
+On the other hand, having a separate application might give you better security 
+guarantees as two processes would be communicating via established binary protocol. 
+Tendermint Core will not have access to application's state. 
+If that is the way you with to proceed, use the [Writing a built-in Tendermint Core application in Go](./go.md) guide instead of this one.
```

Comment:
```suggestion
guarantees as two processes would be communicating via an established binary protocol. 
Tendermint Core will not have access to application's state. 
If that is the way you wish to proceed, use the [Writing a built-in Tendermint Core application in Go](./go.md) guide instead of this one.
```